helix-rest/src/main/java/org/apache/helix/rest/metadatastore/accessor/ZkRoutingDataWriter.java

```diff
@@ -344,6 +346,11 @@ protected boolean deleteZkShardingKey(String realm, String shardingKey) {
   private String constructUrlSuffix(String... urlParams) {
     List<String> allUrlParameters = new ArrayList<>(
         Arrays.asList(MetadataStoreRoutingConstants.MSDS_NAMESPACES_URL_PREFIX, "/", _namespace));
+    String contextUrlPrefix =
+        System.getProperty(MetadataStoreRoutingConstants.MSDS_CONTEXT_URL_PREFIX_KEY);
+    if (contextUrlPrefix != null && !contextUrlPrefix.isEmpty()) {
+      allUrlParameters.add(0, contextUrlPrefix);
+    }
```

Comment:
Add some JavaDoc explaining what you're doing with examples please? This is one of the things that when you look back at it 6 months from now, you'll have to do some digging around to understand what youre doing. Having a comment with an example would help :)

Reply:
Sure thing.